changed.d

```diff
@@ -142,56 +157,231 @@ string getChangeLog(string revRange)
+    auto nextVersionDate = "September 19, 2016";
+    string previousVersion = "Previous version";
+    bool hideTextChanges = false;
+    string revRange;
+
+    auto helpInformation = getopt(
+        args,
+        std.getopt.config.passThrough,
+        "output|o", &outputFile,
+        "date", &nextVersionDate,
+        "version", &nextVersionString,
+        "prev-version", &previousVersion, // this can automatically be detected
+        "no-text", &hideTextChanges);
+
+    if (args.length >= 2)
     {
-        stderr.writeln("Usage: ./changed <revision range>, e.g. ./changed v2.067.1..upstream/stable");
-        return 1;
+        revRange = args[1];
+
+        // extract the previous version
+        auto parts = revRange.split("..");
+        if (parts.length > 1)
+            previousVersion = parts[0];
     }
+    else
+    {
+        writeln("Skipped querying Bugzilla for changes. Please define a revision range e.g ./changed v2.072.2..upstream/stable");
+    }
+
+    auto f = File(outputFile, "w");
+    auto w = f.lockingTextWriter();
+    w.put("Ddoc\n\n");
+    w.formattedWrite("$(CHANGELOG_NAV_LAST %s)\n\n", previousVersion);
+
+    {
+        w.formattedWrite("$(VERSION %s, =================================================,\n\n", nextVersionDate);
+        scope(exit) w.put(")\n");
+
+        if (!hideTextChanges)
+        {
+            // search for raw change files
+            alias Repo = Tuple!(string, "path", string, "headline");
+            auto repos = [Repo("dmd", "Language changes"),
+                          Repo("druntime", "Runtime changes"),
+                          Repo("phobos", "Library changes")];
+
+            auto changedRepos = repos
+                 .map!(repo => Repo(buildPath("..", repo.path, "changelog"), repo.headline))
+                 .filter!(x => x.path.exists)
+                 .map!(x => tuple!("headline", "changes")(x.headline, x.path.readTextChanges.array))
+                 .filter!(x => !x.changes.empty);
+
+            // print the overview headers
+            changedRepos.each!(x => x.changes.writeTextChangesHeader(w, x.headline));
+
+            if (revRange.length)
+                w.put("$(BR)$(BIG $(RELATIVE_LINK2 bugfix-list, List of all bug fixes and enhancements in D $(VER).))\n\n");
+
+            w.put("$(HR)\n\n");
+
+            // print the detailed descriptions
+            changedRepos.each!(x => x.changes.writeTextChangesBody(w, x.headline));
+
+            if (revRange.length)
+                w.put("$(BR)$(BIG $(LNAME2 bugfix-list, List of all bug fixes and enhancements in D $(VER):))\n\n");
+        }
+        else
+        {
+                w.put("$(BR)$(BIG List of all bug fixes and enhancements in D $(VER).)\n\n");
+        }
+
+        // print the entire changelog history
+        if (revRange.length)
+            revRange.getBugzillaChanges.writeBugzillaChanges(w);
+    }
+
+    w.formattedWrite("$(CHANGELOG_NAV_LAST %s)\n", previousVersion);
```

Comment:
Ok let's go with the most simple, working version for now and then continue to extend it :)